src/vos/vos_ts.c

```diff
@@ -308,9 +308,10 @@ vos_ts_set_allocate(struct vos_ts_set **ts_set, uint64_t flags,
 {
 	uint32_t	size;
 	uint64_t	array_size;
+	uint64_t	cond_mask = VOS_COND_FETCH_MASK | VOS_COND_UPDATE_MASK;
 
 	*ts_set = NULL;
-	if (tx_id == NULL)
+	if (tx_id == NULL && (flags & cond_mask) == 0)
```

Comment:
Sorry, I'm not following the ts_in_tx change. @jolivier23, @mchaarawi, more hints?

Reply:
So it's a bit of a hack to limit code changes.   I will think about how best to improve it.   The bottom line is the ts_set is passed around everywhere I need it and contains the flags that were passed into the vos API. 
  Previously, I was not allocating the ts_set outside of a transaction.   This change allocates it but "disabled" just to save the flags.   It's a bit hacky but I would rather move forward and not block Mohamad's patch any further and think about what, if anything, I can do to improve it.   So, the is_in_tx flag is merely a boolean to indicate that we should check/update timestamps.

Reply:
I think I see. Thanks.